improve docs to provide clairity for `watcher_dbt_execution_queue` (#2705)
astronomer/astronomer-cosmos: docs/guides/run_dbt/airflow-worker/watcher-execution-mode.rst

Reviewer: we also have this section. might be good to check if we can re-order the content to be moved up if it brings more value to our users, and consolidate together with the new content we're adding so that we don't repeat information.

.. versionadded:: 1.14.0

In watcher execution mode there are three different "types" of tasks that are executed:

- Producer tasks: execute a ``dbt build`` for the dbt project being rendered and orchestrated with watcher execution mode

> - Watcher tasks: lightweight sensors that wait for the producer task to complete

Reviewer: I guess the naming here could confuse users. The execution mode itself is called Watcher. So both producer and consumer are watcher tasks.

> - Watcher tasks: lightweight sensors that wait for the producer task to complete

Reviewer 2: ```suggestion
- Consumer tasks: (first try) lightweight sensors that wait for the producer task to complete
```

> - Retries: executes a dbt command for a specific node, **only when there is failure for that node**

Reviewer: The header mentions this as "type" of task, but retrese aren't actually a task type, no? It's a just an incremented try_number for the sensor tasks.

> - Retries: executes a dbt command for a specific node, **only when there is failure for that node**

Reviewer 2: ```suggestion
- Consumer tasks: (retries) executes a dbt command for a specific node, **only when there is failure for that node**
```

Producer tasks typically require a high-memory worker to execute the ``dbt build`` command. On their first attempt, the consumer sensors require minimal CPU and memory resources. However, if these tasks retry, they execute the dbt command for the node, which may require significantly more resources. To ensure that the producer tasks and retry attempts have the appropriate resources for execution, the ``watcher_dbt_execution_queue`` configuration can be set.

The ``watcher_dbt_execution_queue`` configuration allows you to specify the worker queue that the producer tasks and retry attempts use. This enables you to:

> - **Optimize resource allocation** — Use high-resource workers for producer tasks, lightweight workers for initial sensor execution, and high-resource workers for retries

Reviewer: ```suggestion
- **Optimize resource allocation** — Use high-resource workers for producer tasks and sensor retries, and lightweight workers for initial sensor execution.
```
There is duplication of the the phrase "high-resource workers" and trying to see if we can avoid that.

- **Improve scheduling efficiency** — Prevent resource contention between producer/retry executions and initial sensor tasks

> - **Scale independently** — Scale "execution" queues (producer and retry) separately from watcher tasks

Reviewer: ```suggestion
- **Scale independently** — Scale "execution" queues (producer and retry) separately from sensor tasks
```
